js/battledata.js

```diff
@@ -826,11 +826,66 @@ var Tools = {
 			}
 			return {tagName: tagName, attribs: attribs};
 		};
+		var localizeTimes = function (tree) {
+			var times = tree.getElementsByTagName('time');
+			for (var i = 0; i < times.length; i++) {
+				var timeElement = times[i];
+				var time = timeElement.textContent;
+
+				// Require ISO 8601 time. While more time formats are supported by
+				// most JavaScript implementations, it isn't required, and how to
+				// exactly enforce ignoring user agent timezone setting is not obvious.
+				// As dates come from the server which isn't aware of client timezone,
+				// a particular timezone is required.
+				//
+				// This regular expression is split into three groups.
+				//
+				// Group 1 - date
+				// Group 2 - time (seconds and milliseconds are optional)
+				// Group 3 - optional timezone
+				//
+				// Group 1 and group 2 are split to allow using space as a separator
+				// instead of T. Stricly speaking ECMAScript 5 specification only
+				// allows T, however it's more practical to also allow spaces.
+				var parts = /^\s*([+-]?\d{4,}-\d{2}-\d{2})[T ](\d{2}:\d{2}(?::\d{2}(?:\.\d{3})?)?)(Z|[+-]\d{2}:\d{2})?\s*$/i.exec(time);
+				if (!parts) continue;
+
+				var parsedTime = new Date(parts[1] + 'T' + parts[2] + (parts[3] || 'Z').toUpperCase());
+				// Very old (pre-ES5) web browsers may be incapable of parsing ISO 8601
+				// dates. In such a case, gracefully continue without replacing the date
+				// format.
+				if (!parsedTime.getTime()) continue;
+
+				var formattedTime;
+				// Try using Intl API if it exists
+				if (window.Intl && window.Intl.DateTimeFormat) {
+					formattedTime = new Intl.DateTimeFormat(undefined, {month: 'long', day: 'numeric', hour: 'numeric', minute: 'numeric'}).format(parsedTime);
+				} else {
+					// toLocaleString even exists in ECMAScript 1, so no need to check
+					// if it exists.
+					formattedTime = parsedTime.toLocaleString();
+				}
+				timeElement.textContent = formattedTime;
+			}
+		};
 		return function (input) {
-			return html.sanitizeWithPolicy(getString(input), tagPolicy);
+			var sanitized = html.sanitizeWithPolicy(getString(input), tagPolicy);
+			// Unfortunately Caja doesn't support modifying tags in case of <time> tag,
+			// so Caja's output is converted into DOM just to modify it. Element name
```

Comment:
### Comparing the rewrite before vs rewrite after approaches
*Rewrite after:*
As long as the rewrite is very especific so as not to change the grammar, there should not be issues, for there is not context where a substring ``<time datetime="2017-02-14">2017-02-14 08:00</time>`` is sintactically or semantically different from ``<time datetime="2017-02-14">2017-02-14 09:30</time>``, so none of them could be safer than the other.

**Limitations**: We could only cover a limited set of attributes for ``time``, and with a rigid grammar. Specifically, stay away from ``style``.

*Rewrite before*
The oddities resulting from regex-replacing "tricky" HTML, such as ``<div title="<time>">`` are already documented in the [pokemonicon discussion](https://github.com/Zarel/Pokemon-Showdown-Client/pull/944#issuecomment-308948392).

In particular, a regex replace without backing from a SAX parser *will* change the grammar and semantics of the final HTML output in unpredictable ways. Of course, we are still left with the guarantees provided by Caja, since Caja can process this wild unknown HTML into a sanitized output.

**Limitations**: A best-effort approach can be followed to cover all sorts of global and ``time``-specific attributes. However, bugs arising from tricky HTML would be unavoidable and their prevalence will correlate with the grammar we want to cover. Likely, stay away from ``style``.